### PR TITLE
snap: add `snap is-managed` for console-conf

### PR DESCRIPTION
**Pull request by mvo5**
For console-conf so that it can check this in its systemd job: https://github.com/CanonicalLtd/subiquity/pull/169

### CODE REVIEW

**Comment by mwhudson** [comment]
:champagne:

File: cmd/snap/cmd_is_managed.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/i18n"
+
+	"github.com/jessevdk/go-flags"
+)
+
+var shortIsManagedHelp = i18n.G("internal")
```
Comment (niemeyer) [inline cmd/snap/cmd_is_managed.go]: Can we please fix these later?  `snap is-managed --help` actually works, even if hidden. Suggestion:
```
var shortIsManagedHelp = i18n.G("Exits zero on managed system")
var longIsManagedHelp = i18n.G(`
The is-managed command will exit with a zero status if snapd has registered users.
`)
```

File: cmd/snap/cmd_is_managed.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+func (cmd cmdIsManaged) Execute(args []string) error {
+	if len(args) > 0 {
+		return ErrExtraArgs
+	}
+
+	is, err := Client().SysInfo()
+	if err != nil {
+		return fmt.Errorf("cannot get system information: %s", err)
```
Comment (niemeyer) [inline cmd/snap/cmd_is_managed.go]: SysInfo itself should return something along those lines in its error message.

File: cmd/snap/cmd_is_managed.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	}
+
+	if is.Managed {
+		fmt.Fprintf(Stdout, "system is managed\n")
```
Comment (niemeyer) [inline cmd/snap/cmd_is_managed.go]: Can we please short these to `managed` and `unmanaged`? There's a subtle detail here about the fact that on classic the _system_ is managed, but we can still claim that _snapd_ is unmanaged.

**Comment by niemeyer** [comment]
Hmm.. perhaps we should go for "snap managed" instead.

**Comment by niemeyer** [comment]
On the other hand, "system is managed" as output is more descriptive on a random script running.

It's also a bit awkward to see an error message when just answering the question, but we need the status.

I may pick this up for sorting these details without you having to worry.

File: cmd/snap/cmd_is_managed.go (hunk 1 above)
Comment (niemeyer) [inline cmd/snap/cmd_is_managed.go]:
```
% snap firstboot -h
Usage:
  snap [OPTIONS] firstboot

internal

```

**Comment by niemeyer** [comment]
Picked up in #2102.

**Pull request closed without merging**